tools/ci_build/github/azure-pipelines/linux-cpu-minimal-build-ci-pipeline.yml

```diff
@@ -66,31 +78,59 @@ jobs:
               --minimal_build \
               --disable_exceptions
       workingDirectory: $(Build.SourcesDirectory)
+
   - task: CmdLine@2
-    displayName: Build minimal onnxruntime and run tests with exceptions enabled
+    displayName: Build minimal onnxruntime with exceptions enabled and type reduction disabled and run tests
     inputs:
       script: |
         docker run --rm \
           --volume $(Build.SourcesDirectory):/onnxruntime_src \
           --volume $(Build.BinariesDirectory):/build \
-          --volume $HOME/.test_data:/home/onnxruntimedev/.test_data \
+          --volume $(test_data_directory):/home/onnxruntimedev/.test_data \
           -e ALLOW_RELEASED_ONNX_OPSET_ONLY=1 \
           -e NIGHTLY_BUILD \
           -e BUILD_BUILDNUMBER \
           onnxruntimecentoscpubuild \
-            /bin/bash /onnxruntime_src/tools/ci_build/github/linux/ort_minimal/build_minimal_ort_and_run_tests.sh
+            /bin/bash /onnxruntime_src/tools/ci_build/github/linux/ort_minimal/build_minimal_ort_and_run_tests.sh \
+              --build-directory /build \
```

Comment:
Nit, can use --build-directory /build/without_type_reduction
Or clear the build at the end of build_minimal_ort_and_run_tests.sh

Reply:
i can use a more specific build directory, good idea. i think we need to keep this build directory around for the later upload test results step though